geode-core/src/main/java/org/apache/geode/internal/cache/wan/AbstractGatewaySender.java

```diff
@@ -1118,6 +1115,24 @@ public void distribute(EnumListenerEvent operation, EntryEventImpl event,
     }
   }
 
+  private void recordDroppedEvent(EntryEventImpl event) {
+    final boolean isDebugEnabled = logger.isDebugEnabled();
+    if (this.eventProcessor != null) {
+      this.eventProcessor.registerEventDroppedInPrimaryQueue(event);
+    } else {
+      // Add empty event so that in case the event stays for long in
+      // tmpDroppedEvents it takes as little space as possible.
+      // No need to have all the contents of the event for a dropped one.
+      EntryEventImpl emptyEvent = new EntryEventImpl(event.getKey(), false);
```

Comment:
Again, don't clone a new event. In the new code change, the "else" will only be executed in very very small time windows. 

You have proved that when the sender is shutdown, the eventProcessor will be kept. So enqueue will not go to the "else". We will never worry about the tmpDroppedEvents become too big.

Reply:
ok. I will remove it.
Anyway, if the gateway sender is created with manualStart=true then the eventProcessor will not be yet instantiated. In that case, and until the gateway sender is started we could see dropped events put in tmpDroppedEvents. It cannot be guaranteed that it will be a small time window.

Reply:
For this part, it's unnecessary to enhance registerEventDroppedInPrimaryQueue() since it's used for a very small window (sender is restarting). 

I feel your previous solution is good enough.